### ISSUE TITLE
MTK6768: Continuous battery consumption during idel/sleep state without any background activity

### ISSUE THREAD

**Issue by abdullah5490** [phhusson/treble_experimentations#2233]
I tried V3.xx and now trying my luck with 4.xx with same results. without any app/activity, device is consuming 3/4% per hour while no notification, no screen turned on, no call came in. Its a huge difference from stock.
Look at the screenshot:
![Screenshot_20220320-002157_Pixel Launcher](https://user-images.githubusercontent.com/15821394/159136106-ea3dcef2-53ba-4206-a4a1-f8241ccaad05.png)
I didn't used this phone during last 24hrs. Even then the battery drops from 50% to 4% during this time, and "phone idel" usage is 19%.

Here are the logs
[radio-infinix-x690b.txt](https://github.com/phhusson/treble_experimentations/files/8309989/radio-infinix-x690b.txt)
:
[logcat-infinix-x690b.txt](https://github.com/phhusson/treble_experimentations/files/8309988/logcat-infinix-x690b.txt)

**Comment by phhusson** [comment]
I'm not seeing anything weird in the logs, except a log of logs from TTS, but I guess you're using a technology for the visually impaired?

Anyway, let your device go to sleep, screen off, don't touch it, and don't receive any notifications for 30 minutes, and do
`adb shell 'dumpsys  > /sdcard/dumpsys.txt'` and attach dumpsys.txt from your phone internal storage here.

**Comment by abdullah5490** [comment]
@phhusson yes I'm blind and use screen reader to operate phone/computer. This is screen reader for which I need android11/12 that is not available natively for this device and I can't afford to buy a new phone just for software change.
I left my phone at 8:42am, plugged-in USB cable at 9:12am and pulled  the dumpsys via adb shell at 9:14am.
[dumpsys.txt](https://github.com/phhusson/treble_experimentations/files/8310602/dumpsys.txt)
According to latest usage stats of last 24hrs, the "phone idel" went to 30% followed by "mobile network" of 25% where screen on time is only 19Minutes.

**Comment by abdullah5490** [comment]
@phhusson i understand you might be busy, but still i believe that you have checked my dumpsys till now. so, is there any help for me?

**Comment by phhusson** [comment]
Your dumpsys is severely short, it is missing 95% of the infos.

" I left my phone at 8:42am, plugged-in USB cable at 9:12am and pulled the dumpsys via adb shell at 9:14am." ====> What was the battery level after that?

**Comment by abdullah5490** [comment]
> Your dumpsys is severely short, it is missing 95% of the infos.
So i've used exactly the same process you told above, if anything else i could do to make this dumpsys comprehensive, let me know and i'm ready to provide again. 
 
> " I left my phone at 8:42am, plugged-in USB cable at 9:12am and pulled the dumpsys via adb shell at 9:14am." ====> What was the battery level after that?
Since it was more than 5days ago so i don't remember the percentage, but i remember it was dropped 2% during those 30mins sleep.

**Comment by phhusson** [comment]
I have no idea how the dumpsys can be too short. Maybe you didn't wait for the command to finish? I don't know

To know it is complete, you must have a line with `DUMP OF SERVICE window:`

You can see at the start of the file that it lists the services, DockObserver, SurfaceFlinger..., and there are various `DUMP OF SERVICE xxxx:` lines, and the latest `DUMP OF SERVICE` is `android.security.identity` which is service number 13 while there are 215 services listed

**Comment by abdullah5490** [comment]
I understand what you said and allow me to try once again with another dumpsys file in next comment. Here's what I did, launched an adb shell first, then executed the command to create the dump file in /sdcard/. Without opening an adb shell windows separately, the full command doesn't work. 
> I have no idea how the dumpsys can be too short. Maybe you didn't wait for the command to finish? I don't know
> 
> 
> 
> To know it is complete, you must have a line with `DUMP OF SERVICE window:`
> 
> 
> 
> You can see at the start of the file that it lists the services, DockObserver, SurfaceFlinger..., and there are various `DUMP OF SERVICE xxxx:` lines, and the latest `DUMP OF SERVICE` is `android.security.identity` which is service number 13 while there are 215 services listed

**Comment by abdullah5490** [comment]
@phhusson ok, here is the complete dump this time (i've varified).
Left phone at 7:20 with 97% of battery, turned the screen on at 7:51, battery left 95%. No notification, no screen popped-up during these 31mins.
[dumpsys.txt](https://github.com/phhusson/treble_experimentations/files/8351653/dumpsys.txt)

There were 4 warnings  during dumping process regarding "can't find service {service name}" and few errors saying  "error dumping info for {service name}"

**Comment by phhusson** [comment]
Right the error are expected.

So, looking at yours log, there are two things preventing deep sleep:
```
  PARTIAL_WAKE_LOCK              '*job*/ru.yandex.disk/.service.DiskJobService' ACQ=-1m41s42ms LONG (uid=1000 pid=1304 ws=WorkSource{10226})
  DOZE_WAKE_LOCK                 'DreamManagerService' ACQ=-59s351ms (uid=1000 pid=1304)
```


Looks like the Yandex Disk app is too busy scanning your internal storage to let your device sleep. If you want to keep this app, install an app freezer so it doesn't eat battery in backgrond. For instance you can use SuperFreezz app.

Also it looks like you have Always-on display enabled eating battery, not sure why. Since you're blind, I'm guessing it's useless, so disable it. It's in Settings => Display => Lockscreen

**Comment by abdullah5490** [comment]
@phhusson so after deleting "yandex.disk" and disabling AOD, its improved allot with the ratio of 50%. Now, its dropping 1% after every hour (improved from 2% per hour previously).
I'm submitting latest dump and request your kindness to please have a look again to determine the current status of AOD, and pointing out next possible cause of battery usage during sleep.
[dumpsys1.txt](https://github.com/phhusson/treble_experimentations/files/8364610/dumpsys1.txt)
With your help, I hope its possible to stop unusual battery usage completely as it is improved too much after applying just 1 suggestion came from your side.
Also (if possible), check that  all functions of device's overlay are working properly or if they need any further adjustments.

**Comment by abdullah5490** [comment]
@phhusson i'm still waiting for your kind response on above comment, as the situation improved allot due by your help, and i'm committed to make more improvements!

**Comment by phhusson** [comment]
Uh sorry, what kind of answer are you expecting? You have 4 days of battery life, that sounds pretty good to me?

**Comment by abdullah5490** [comment]
> Uh sorry, what kind of answer are you expecting? 
After applying your suggestions which u gave after analysing dump, the battery during sleep was improved. Then i've submitted a fresh dump afterward and expecting another suggestion after checking for remaining cause  
> You have 4 days of battery life, that sounds pretty good to me?
4days life of only standby/sleep without usage. Because I didn't installed any app in fearof battery drain during sleep, as one of the app was previously causing, and i don't know how to trace-out such app which is preventing deep sleep. Moreover, i was also waiting for your comment after my fresh dump above, and want to move step by step in improvement of sleep consumption.

**Comment by abdullah5490** [comment]
@phhusson 
> Uh sorry, what kind of answer are you expecting?
After applying your suggestions above, the battery during sleep was improved (not perfect). Then i've submitted a fresh dump afterward and expecting another suggestion after checking for remaining cause
> You have 4 days of battery life, that sounds pretty good to me?
4days life of only standby/sleep without usage. Because I didn't installed any app in fearof battery drain during sleep, as one of the app was previously causing, and i don't know how to trace-out such app which is preventing deep sleep. Moreover, i was also waiting for your comment after my fresh dump above, and want to move step by step in improvement of sleep consumption.

**Comment by abdullah5490** [comment]
@phhusson Hello. Allow me to explain another observation about battery consumption in idel.
When I disable sim cards and keep wifi enabled, the battery usage is greatly decrease, came back to normal just like stock. 
When I keep wifi and mobile data off, but sim cards are enabled for calls/messages only, the idel battery consumption is really high even without any call or SMS take place.
It means something related to simcards is effecting the battery during idel. 
Any suggestion to have a deap look and improve this situation? 
Using V413-AOSP12.1
